wiki_doc

```diff
@@ -0,0 +1,190 @@
+## Valkey Commands
+
+For information on the supported commands and their corresponding parameters, we recommend referring to the documentation in the code. This documentation provides in-depth insights into the usage and options available for each command.
+
+### Command String Arguments
+
+Valkey stores strings as sequences of bytes, so that may include text, serialized objects, or binary arrays. As such, to pass Valkey strings as arguments to commands, or receive Valkey strings in responses, GLIDE offers `GlideString`, which represents two APIs:
+
+1. `string`: an UTF-8 string.
+2. `Buffer`: an array of bytes.
+
+```typescript
+/**
+ * Union type that can store either a valid UTF-8 string or array of bytes.
+ */
+export type GlideString = string | Buffer;
+```
+
+The return types of command API:
+
+-   Command uses `string` as the return type if it constains a specific string value, which could never be binary data. For example, `type` command returns one of pre-defined strings, or `set` command which returns `OK`.
+-   Command uses `Buffer` as the return type, because it always contains binary data. For example, `dump` and `functionDump` commands.
+-   Most commands use `GlideString` as the common return type.
+
+### Decoder
+
+In Node.js, most users expect the responses to be in string format unless specified otherwise. Users can select which data type they expect to receive as a response. The user has the option of converting the response into either a string or bytes for each command. In addition, as part of the client creation process, the user has the option of setting a default for all commands if no decoder argument is passed to the command.
+
+`Decoder` is an optional parameter for all commands and it has two options
```

Comment:
link to the definition: https://github.com/valkey-io/valkey-glide/blob/main/node/src/BaseClient.ts#L283